.github/workflows/ci-build.yaml

```
@@ -151,57 +151,39 @@ jobs:
         env:
           CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
 
-  argoexec-image:
-    name: argoexec-image
+  argo-images:
+    name: argo-images
     # needs: [ lint ]
     runs-on: ubuntu-latest
     timeout-minutes: 10
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - image: argoexec
+          - image: argocli
     steps:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
       - uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
       - name: Build and export
         uses: docker/build-push-action@4a13e500e55cf31b7a5d59a38ab2040ab0f42f56 # v5.1.0
         with:
           context: .
-          tags: quay.io/argoproj/argoexec:latest
-          outputs: type=docker,dest=/tmp/argoexec_image.tar
-          target: argoexec
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
-      - name: Upload
-        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
-        with:
-          name: argoexec
-          path: /tmp/argoexec_image.tar
-          if-no-files-found: error
-
-  argocli-image:
-    name: argocli-image
-    # needs: [ lint ]
-    runs-on: ubuntu-latest
-    timeout-minutes: 15
-    steps:
-      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
-      - uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
-      - name: Build and export
-        uses: docker/build-push-action@4a13e500e55cf31b7a5d59a38ab2040ab0f42f56 # v5.1.0
-        with:
-          context: .
-          tags: quay.io/argoproj/argocli:latest
-          outputs: type=docker,dest=/tmp/argocli_image.tar
-          target: argocli
+          tags: quay.io/argoproj/${{matrix.image}}:latest
+          outputs: type=docker,dest=/tmp/${{matrix.image}}_image.tar
+          target: ${{matrix.image}}
           cache-from: type=gha
           cache-to: type=gha,mode=max
       - name: Upload
-        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 #v4.3.3
         with:
-          name: argocli
-          path: /tmp/argocli_image.tar
+          name: ${{matrix.image}}_image.tar
```

```suggestion
          name: ${{matrix.image}}
```

From the previous code, this should just be the name of the image. Not sure if this was a copy+paste typo?

---

It's deliberate. We're pattern matching this with `pattern: '*_image.tar'` on line 164. This pattern would be `'*'` without the extra parts to the name.

---

Yes, I thought that would suffice, no? Or were you trying to be extra careful in case at some point other artifacts were added or something?